web/components/admin/AdminGate/index.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Redirect } from 'react-router-dom';
+import { firebaseConnect, firestoreConnect, isLoaded, isEmpty } from 'react-redux-firebase';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import { Logout } from '../../auth';
+
+const AdminGate = ({ auth, firestore, children }) => {
+  const {
+    status: {
+      requesting: {
+        admins: isAdminsLoading,
+      },
+    },
+  } = firestore;
+
+  if (!isLoaded(auth) || isAdminsLoading) {
+    return (<span>Loading...</span>);
+  }
+
+  const {
+    data: {
+      admins,
+    },
+  } = firestore;
+
+  const { uid } = auth;
+  const isAdmin = uid in admins;
```

Comment:
`uid in admins` is throwing an error when `uid` is `undefined` and `admins` is `null` (i.e. when you're not logged in and try accessing any of the admin routes).

Specifically: `Uncaught TypeError: Cannot use 'in' operator to search for 'undefined' in null`

Reply:
o good catch

Reply:
we don't need to check whether `uid` is undefined/null though, because `null in {...}` is false

Reply:
Oh true